thirdparty/libbacktrace/read.c

```diff
@@ -74,7 +69,13 @@ backtrace_get_view (struct backtrace_state *state, int descriptor,
   got = 0;
   while (got < size)
     {
-      r = read (descriptor, view->base, size - got);
+/* GODOT start */
+      int sz = size - got;
+      if (size - got > INT_MAX) {
+        sz = INT_MAX;
+      }
```

Comment:
Should we use the same type as `size` and `got`?
```suggestion
      uint64_t sz = size - got;
      if (sz > INT_MAX) {
        sz = INT_MAX;
      }
```

Reply:
I think this checks for 2s complement overflow. In a uint64_t this can't happen. I don't think your suggestion is correct.

Reply:
nevermind, that doesn't make a lot of sense. I retract my comment.

Reply:
It's one check less, but should work the same. Both `size` and `got` are `uint64_t`.